core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -0,0 +1,144 @@
+      }
+    }
+  }
+
+  public HadoopCatalog(Configuration conf) {
+    this(conf, null);
+  }
+
+  @Override
+  public org.apache.iceberg.Table createTable(
+      TableIdentifier identifier, Schema schema, PartitionSpec spec, Map<String, String> properties) {
+    Preconditions.checkArgument(identifier.namespace().levels().length == 1,
+            "Missing database in table identifier: %s", identifier);
+    Path tablePath = new Path(defaultWarehouseLocation(identifier));
+    try {
+      FileSystem fs = Util.getFs(tablePath, conf);
+      if (!fs.isDirectory(tablePath)) {
+        fs.mkdirs(tablePath);
+      } else {
+        throw new AlreadyExistsException("the table already exists: " + identifier);
+      }
+    } catch (IOException e) {
+      throw new RuntimeIOException("failed to create directory", e);
+    }
+    return super.createTable(identifier, schema, spec, null, properties);
+  }
+
+  public org.apache.iceberg.Table createTable(
+          TableIdentifier identifier, Schema schema, PartitionSpec spec) {
+    Preconditions.checkArgument(identifier.namespace().levels().length == 1,
+            "Missing database in table identifier: %s", identifier);
+    return createTable(identifier, schema, spec, null, null);
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier identifier) {
+    Preconditions.checkArgument(identifier.namespace().levels().length == 1,
+                "Missing database in table identifier: %s", identifier);
+    return new HadoopTableOperations(new Path(defaultWarehouseLocation(identifier)), conf);
+  }
+
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
+    String dbName = tableIdentifier.namespace().level(0);
+    String tableName = tableIdentifier.name();
+    return this.warehouseUri + "/" + dbName + ".db" + "/" + tableName;
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    Preconditions.checkArgument(identifier.namespace().levels().length == 1,
+                "Missing database in table identifier: %s", identifier);
+
+    Path tablePath = new Path(defaultWarehouseLocation(identifier));
+    TableOperations ops = newTableOps(identifier);
+    TableMetadata lastMetadata;
+    if (purge && ops.current() != null) {
+      lastMetadata = ops.current();
+    } else {
+      lastMetadata = null;
+    }
+
+    FileSystem fs = Util.getFs(tablePath, conf);
+    try {
+      if (purge && lastMetadata != null) {
+        dropTableData(ops.io(), lastMetadata);
+      }
+      fs.delete(tablePath, true /* recursive */);
+      return true;
+    } catch (IOException e) {
+      throw new RuntimeIOException(e, "Failed to delete file: %s", tablePath);
+    }
+  }
+
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+    throw new RuntimeException("rename a hadoop table is not supported");
```

Comment:
This should be `UnsupportedOperationException`. Also, the message could be a bit more clear. What about "Cannot rename Hadoop tables."